xcc/connection.py

```diff
@@ -241,6 +241,11 @@ def request(self, method: str, path: str, **kwargs) -> requests.Response:
             self.update_access_token()
             response = self._request(method=method, url=url, headers=self.headers, **kwargs)
 
+        if kwargs.get("stream", False) is True:
+            # Avoid eagerly fetching the content for streaming requests.
+            response.raise_for_status()
```

Comment:
Why the `raise_for_status()` here and not just returning the response object?

Reply:
Good question! I defer to the docstring of this function which states:
```
Raises:
    requests.exceptions.RequestException: if there was an issue sending
        the HTTP request or the status code of the HTTP response
        indicates that an error occurred (i.e., 4XX or 5XX)
```
Put another way, this function promises to raise a RequestException if the status code of a HTTP request is 4XX or 5XX so that callers don't have to do this themselves.